torch/csrc/distributed/c10d/ProcessGroupNCCL.cpp

```diff
@@ -854,6 +897,27 @@ void ProcessGroupNCCL::broadcastUniqueNCCLID(
   }
 }
 
+
+void ProcessGroupNCCL::destroyNCCLComms(const std::string& devNCCLCommMapKey) {
+  if (devNCCLCommMap_.find(devNCCLCommMapKey) == devNCCLCommMap_.end()) {
+    TORCH_INTERNAL_ASSERT(
+        false,
+        "Expected to find key ",
+        devNCCLCommMapKey,
+        " in NCCL communicator map.");
+  }
+  std::vector<std::shared_ptr<NCCLComm>>& ncclComms =
+      devNCCLCommMap_[devNCCLCommMapKey];
+  // Loop through communicators and call ncclCommAbort.
+  for (const auto& comm : ncclComms) {
+    // ncclCommDestroy(comm->getNcclComm()) results in segfault when PG is being
```

Comment:
Could you elaborate a bit more on why this happens?

Reply:
Some segfault issues are mentioned on L43 here: https://caffe2.ai/doxygen-c/html/nccl_8cpp_source.html, as well as in PT - https://github.com/pytorch/pytorch/blob/master/torch/csrc/cuda/nccl.cpp#L373. 

Although, it is a bit strange since this isn't called in dtor but still getting a segfault (in `~ProcessGroupNCCL`) when enabling this, and calling `ncclCommAbort` seems to have fixed the issue.